base/common/src/main/java/com/netscape/certsrv/profile/ProfileData.java

```diff
@@ -273,18 +286,339 @@ public boolean equals(Object obj) {
+                                    attributeElement.setAttribute("name", attribute.getName());
+                                }
+                                if (attribute.getValue() != null) {
+                                    Element ppValueElement = document.createElement("Value");
+                                    ppValueElement.appendChild(document.createTextNode(attribute.getValue()));
+                                    attributeElement.appendChild(ppValueElement);
+                                }
+                                Descriptor descriptor = attribute.getDescriptor();
+                                if (descriptor != null) {
+                                    Element descriptorElement = document.createElement("Descriptor");
+                                    if (descriptor.getSyntax() != null) {
+                                        Element syntaxElement = document.createElement("mSyntax");
+                                        syntaxElement.appendChild(document.createTextNode(descriptor.getSyntax()));
+                                        descriptorElement.appendChild(syntaxElement);
+                                    }
+                                    if (descriptor.getConstraint() != null) {
+                                        Element constraintElement = document.createElement("mConstraint");
+                                        constraintElement.appendChild(document.createTextNode(descriptor.getConstraint()));
+                                        descriptorElement.appendChild(constraintElement);
+                                    }
+                                    if (descriptor.getDescription(Locale.getDefault()) != null) {
+                                        Element descriptionElement = document.createElement("mDescription");
+                                        descriptionElement.appendChild(document.createTextNode(descriptor.getDescription(Locale.getDefault())));
+                                        descriptorElement.appendChild(descriptionElement);
+                                    }
+                                    if (descriptor.getDefaultValue() != null) {
+                                        Element defaultValueElement = document.createElement("mDef");
+                                        defaultValueElement.appendChild(document.createTextNode(descriptor.getDefaultValue()));
+                                        descriptorElement.appendChild(defaultValueElement);
+                                    }
+                                    attributeElement.appendChild(descriptorElement);
+                                }
+                                defElement.appendChild(attributeElement);
+                            }
+                        }
+                        if (!pp.getDef().getParams().isEmpty()) {
+                            for (ProfileParameter param : pp.getDef().getParams()) {
```

Comment:
Same thing with the `if-statement` for params.

Reply:
Agreed, fixed in next revision.